SNOW-352097: ExternalBrowser authentication type does not work for .NET Core projects on Windows, Linux or Mac

## Issue Description
When using ExternalBrowser authentication type, URL to the SSO provider is constructed and a browser window is opened using default OS browser. The code to start it is in Snowflake.Data.Core.Authenticator.ExternalBrowserAuthenticator.StartBrowser
https://github.com/snowflakedb/snowflake-connector-net/blob/master/Snowflake.Data/Core/Authenticator/ExternalBrowserAuthenticator.cs#L156-L174
```CS
        private static void StartBrowser(string url)
        {
            // The following code is learnt from https://brockallen.com/2016/09/24/process-start-for-urls-on-net-core/
#if NETFRAMEWORK
            // .net standard would pass here
            Process.Start(url);
#else
            // hack because of this: https://github.com/dotnet/corefx/issues/10361
            if (RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
            {
                url = url.Replace("&", "^&");
                Process.Start(new ProcessStartInfo("cmd", $"/c start {url}") { CreateNoWindow = true });
            }
            else
            {
                throw new SnowflakeDbException(SFError.UNSUPPORTED_PLATFORM);
            }
#endif
        }
```

The logic of the https://github.com/snowflakedb/snowflake-connector-net/blob/master/Snowflake.Data/Core/Authenticator/ExternalBrowserAuthenticator.cs#L159 is incorrect. 

The `NETFRAMEWORK` preprocessor symbol is used to conditionally compile in `Process.Start(url)` for what the comments suggest are .NET Standard. 
```cs
#if NETFRAMEWORK
            // .net standard would pass here
            Process.Start(url);
```

However, per https://docs.microsoft.com/en-us/dotnet/standard/frameworks#how-to-specify-a-target-framework, `NETFRAMEWORK` preprocessor symbol is for ".NET Framework", not 'NET Standard". 

## Looking at Result of Compilation

When compiled, the DLL in the `C:\snowflake\snowflake-connector-net\Snowflake.Data\bin\Release\net472\Snowflake.Data.dll` as well as in the public nuget package `C:\Users\Daniel\Downloads\snowflake.data.2.0.0\lib\net472\Snowflake.Data.dll` that is targeting .NET Full Framework for Windows Only desktops, which works fine on Windows:

```CS
// Snowflake.Data.Core.Authenticator.ExternalBrowserAuthenticator
using System.Diagnostics;

private static void StartBrowser(string url)
{
    Process.Start(url);
}
```

Looking at the DLL that targets .NET Core in `C:\snowflake\snowflake-connector-net\Snowflake.Data\bin\Release\netstandard2.0\Snowflake.Data.dll` and in `C:\Users\Daniel\Downloads\snowflake.data.2.0.0\lib\netstandard2.0\Snowflake.Data.dll`, that would be used by code running on Mac and Linux, we can see that code that could only run on Windows (and it does run beautifully there), but throws an SnowflakeDbException for Mac and Linux:

```cs
// Snowflake.Data.Core.Authenticator.ExternalBrowserAuthenticator
using Snowflake.Data.Client;
using System.Diagnostics;
using System.Runtime.InteropServices;

private static void StartBrowser(string url)
{
    if (RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
    {
        url = url.Replace("&", "^&");
        Process.Start(new ProcessStartInfo("cmd", $"/c start {url}")
        {
            CreateNoWindow = true
        });
        return;
    }
    throw new SnowflakeDbException(SFError.UNSUPPORTED_PLATFORM);
}
```

## Possible Solution 1
This seems like an honest mistake, and can most easily be fixed by changing to correct preprocessor symbol `NETSTANDARD2_1` or better yet `NET`:
```cs
#if NET
            // .net standard would pass here
            Process.Start(url);
```

## Possible Solution 2
Better yet, change the code to explicitly address 3 different OS versions. In fact https://brockallen.com/2016/09/24/process-start-for-urls-on-net-core/ that is mentioned in comments does just that.

Upon further testing, the ExternalBrowser authentication does not work on any platform. 
using this connection string:
```cs
            string connectionString = String.Format(
                "host={0};account={1};role={2};db={3};schema={4};warehouse={5};user={6};authenticator={7}",
                "instacart.snowflakecomputing.com",
                "instacart",
                "ACCOUNTADMIN",
                "SANDBOX_DB",
                "DANIELODIEVICH",
                "SNOWFLAKE_SUPPORT_WH",
                "dodievich", 
                "externalbrowser");
```

## Windows
conn.Open produces this: 
```
[2021-05-03 14:50:14,763] [1] [DEBUG] [Snowflake.Data.Client.SnowflakeDbConnection] Open Connection.
[2021-05-03 14:50:14,772] [1] [INFO ] [Snowflake.Data.Core.SFSessionProperties] Start parsing connection string.
[2021-05-03 14:50:14,775] [1] [INFO ] [Snowflake.Data.Core.SFSessionProperties] Connection property: HOST, value: instacart.snowflakecomputing.com
[2021-05-03 14:50:14,775] [1] [INFO ] [Snowflake.Data.Core.SFSessionProperties] Connection property: ACCOUNT, value: instacart
[2021-05-03 14:50:14,776] [1] [INFO ] [Snowflake.Data.Core.SFSessionProperties] Connection property: ROLE, value: ACCOUNTADMIN
[2021-05-03 14:50:14,776] [1] [INFO ] [Snowflake.Data.Core.SFSessionProperties] Connection property: DB, value: SANDBOX_DB
[2021-05-03 14:50:14,776] [1] [INFO ] [Snowflake.Data.Core.SFSessionProperties] Connection property: SCHEMA, value: DANIELODIEVICH
[2021-05-03 14:50:14,776] [1] [INFO ] [Snowflake.Data.Core.SFSessionProperties] Connection property: WAREHOUSE, value: SNOWFLAKE_SUPPORT_WH
[2021-05-03 14:50:14,776] [1] [INFO ] [Snowflake.Data.Core.SFSessionProperties] Connection property: USER, value: dodievich
[2021-05-03 14:50:14,776] [1] [INFO ] [Snowflake.Data.Core.SFSessionProperties] Connection property: AUTHENTICATOR, value: externalbrowser
[2021-05-03 14:50:14,782] [1] [DEBUG] [Snowflake.Data.Core.SFSessionProperties] Sesssion property PORT set to default value: 443
[2021-05-03 14:50:14,783] [1] [DEBUG] [Snowflake.Data.Core.SFSessionProperties] Sesssion property SCHEME set to default value: https
[2021-05-03 14:50:14,783] [1] [DEBUG] [Snowflake.Data.Core.SFSessionProperties] Sesssion property CONNECTION_TIMEOUT set to default value: 0
[2021-05-03 14:50:14,783] [1] [DEBUG] [Snowflake.Data.Core.SFSessionProperties] Sesssion property VALIDATE_DEFAULT_PARAMETERS set to default value: true
[2021-05-03 14:50:14,785] [1] [DEBUG] [Snowflake.Data.Core.SFSession] Open Session
[2021-05-03 14:50:14,798] [1] [INFO ] [Snowflake.Data.Core.Authenticator.ExternalBrowserAuthenticator] External Browser Authentication
[2021-05-03 14:50:14,867] [1] [DEBUG] [Snowflake.Data.Core.Authenticator.ExternalBrowserAuthenticator] Get IdpUrl and ProofKey
[2021-05-03 14:50:16,321] [1] [DEBUG] [Snowflake.Data.Core.Authenticator.ExternalBrowserAuthenticator] Open browser
[2021-05-03 14:50:17,826] [1] [ERROR] [Snowflake.Data.Client.SnowflakeDbConnection] Unable to connect
System.ComponentModel.Win32Exception (2): The system cannot find the file specified.
   at System.Diagnostics.Process.StartWithCreateProcess(ProcessStartInfo startInfo)
   at System.Diagnostics.Process.Start()
   at System.Diagnostics.Process.Start(ProcessStartInfo startInfo)
   at System.Diagnostics.Process.Start(String fileName)
   at Snowflake.Data.Core.Authenticator.ExternalBrowserAuthenticator.StartBrowser(String url) in C:\snowflake\snowflake-connector-net\Snowflake.Data\Core\Authenticator\ExternalBrowserAuthenticator.cs:line 161
   at Snowflake.Data.Core.Authenticator.ExternalBrowserAuthenticator.Snowflake.Data.Core.Authenticator.IAuthenticator.Authenticate() in C:\snowflake\snowflake-connector-net\Snowflake.Data\Core\Authenticator\ExternalBrowserAuthenticator.cs:line 112
   at Snowflake.Data.Core.SFSession.Open() in C:\snowflake\snowflake-connector-net\Snowflake.Data\Core\SFSession.cs:line 135
   at Snowflake.Data.Client.SnowflakeDbConnection.Open() in C:\snowflake\snowflake-connector-net\Snowflake.Data\Client\SnowflakeDbConnection.cs:line 108
[2021-05-03 14:50:18,837] [1] [DEBUG] [Snowflake.Data.Client.SnowflakeDbConnection] Close Connection.
```

That is coming straight from https://github.com/snowflakedb/snowflake-connector-net/blob/master/Snowflake.Data/Core/Authenticator/ExternalBrowserAuthenticator.cs#L161

## Mac
On Mac the situation is identical
```
[2021-05-03 15:01:55,986] [1] [DEBUG] [Snowflake.Data.Client.SnowflakeDbConnection] Open Connection.
[2021-05-03 15:01:55,995] [1] [INFO ] [Snowflake.Data.Core.SFSessionProperties] Start parsing connection string.
[2021-05-03 15:01:55,998] [1] [INFO ] [Snowflake.Data.Core.SFSessionProperties] Connection property: HOST, value: instacart.snowflakecomputing.com
[2021-05-03 15:01:55,998] [1] [INFO ] [Snowflake.Data.Core.SFSessionProperties] Connection property: ACCOUNT, value: instacart
[2021-05-03 15:01:55,998] [1] [INFO ] [Snowflake.Data.Core.SFSessionProperties] Connection property: ROLE, value: ACCOUNTADMIN
[2021-05-03 15:01:55,998] [1] [INFO ] [Snowflake.Data.Core.SFSessionProperties] Connection property: DB, value: SANDBOX_DB
[2021-05-03 15:01:55,998] [1] [INFO ] [Snowflake.Data.Core.SFSessionProperties] Connection property: SCHEMA, value: DANIELODIEVICH
[2021-05-03 15:01:55,998] [1] [INFO ] [Snowflake.Data.Core.SFSessionProperties] Connection property: WAREHOUSE, value: SNOWFLAKE_SUPPORT_WH
[2021-05-03 15:01:55,998] [1] [INFO ] [Snowflake.Data.Core.SFSessionProperties] Connection property: USER, value: dodievich
[2021-05-03 15:01:55,998] [1] [INFO ] [Snowflake.Data.Core.SFSessionProperties] Connection property: AUTHENTICATOR, value: externalbrowser
[2021-05-03 15:01:56,001] [1] [DEBUG] [Snowflake.Data.Core.SFSessionProperties] Sesssion property PORT set to default value: 443
[2021-05-03 15:01:56,001] [1] [DEBUG] [Snowflake.Data.Core.SFSessionProperties] Sesssion property SCHEME set to default value: https
[2021-05-03 15:01:56,002] [1] [DEBUG] [Snowflake.Data.Core.SFSessionProperties] Sesssion property CONNECTION_TIMEOUT set to default value: 0
[2021-05-03 15:01:56,002] [1] [DEBUG] [Snowflake.Data.Core.SFSessionProperties] Sesssion property VALIDATE_DEFAULT_PARAMETERS set to default value: true
[2021-05-03 15:01:56,003] [1] [DEBUG] [Snowflake.Data.Core.SFSession] Open Session
[2021-05-03 15:01:56,104] [1] [INFO ] [Snowflake.Data.Core.Authenticator.ExternalBrowserAuthenticator] External Browser Authentication
[2021-05-03 15:01:56,418] [1] [DEBUG] [Snowflake.Data.Core.Authenticator.ExternalBrowserAuthenticator] Get IdpUrl and ProofKey
[2021-05-03 15:01:57,452] [1] [DEBUG] [Snowflake.Data.Core.Authenticator.ExternalBrowserAuthenticator] Open browser
[2021-05-03 15:01:57,456] [1] [ERROR] [Snowflake.Data.Client.SnowflakeDbConnection] Unable to connect
System.ComponentModel.Win32Exception (2): No such file or directory
   at System.Diagnostics.Process.ForkAndExecProcess(String filename, String[] argv, String[] envp, String cwd, Boolean redirectStdin, Boolean redirectStdout, Boolean redirectStderr, Boolean setCredentials, UInt32 userId, UInt32 groupId, UInt32[] groups, Int32& stdinFd, Int32& stdoutFd, Int32& stderrFd, Boolean usesTerminal, Boolean throwOnNoExec)
   at System.Diagnostics.Process.StartCore(ProcessStartInfo startInfo)
   at System.Diagnostics.Process.Start()
   at System.Diagnostics.Process.Start(ProcessStartInfo startInfo)
   at System.Diagnostics.Process.Start(String fileName)
   at Snowflake.Data.Core.Authenticator.ExternalBrowserAuthenticator.StartBrowser(String url) in C:\snowflake\snowflake-connector-net\Snowflake.Data\Core\Authenticator\ExternalBrowserAuthenticator.cs:line 161
   at Snowflake.Data.Core.Authenticator.ExternalBrowserAuthenticator.Snowflake.Data.Core.Authenticator.IAuthenticator.Authenticate() in C:\snowflake\snowflake-connector-net\Snowflake.Data\Core\Authenticator\ExternalBrowserAuthenticator.cs:line 112
   at Snowflake.Data.Core.SFSession.Open() in C:\snowflake\snowflake-connector-net\Snowflake.Data\Core\SFSession.cs:line 135
   at Snowflake.Data.Client.SnowflakeDbConnection.Open() in C:\snowflake\snowflake-connector-net\Snowflake.Data\Client\SnowflakeDbConnection.cs:line 108
[2021-05-03 15:01:57,640] [1] [DEBUG] [Snowflake.Data.Client.SnowflakeDbConnection] Close Connection.
```

So I must stand just a bit corrected, my previous errors were with privately built copy of this where I changes project file targets from `netstandard2.0;net472` to `net5` which negates this:
```xml
  <PropertyGroup Condition="$([System.Text.RegularExpressions.Regex]::IsMatch('$(TargetFramework)', '^net\d'))">
	<DefineConstants>NETFRAMEWORK</DefineConstants>
  </PropertyGroup>
```

If I run the code with public assembly from nuget, the errors on Mac and Linux are slightly different:
```
[2021-05-04 08:44:18,195] [1] [DEBUG] [Snowflake.Data.Core.SFSession] Open Session
[2021-05-04 08:44:18,201] [1] [INFO ] [Snowflake.Data.Core.Authenticator.ExternalBrowserAuthenticator] External Browser Authentication
[2021-05-04 08:44:18,247] [1] [DEBUG] [Snowflake.Data.Core.Authenticator.ExternalBrowserAuthenticator] Get IdpUrl and ProofKey
[2021-05-04 08:44:18,862] [1] [DEBUG] [Snowflake.Data.Core.Authenticator.ExternalBrowserAuthenticator] Open browser
[2021-05-04 08:44:18,868] [1] [ERROR] [Snowflake.Data.Client.SnowflakeDbConnection] Unable to connect
Error: OS platform is not supported. SqlState: , VendorCode: 270012, QueryId: 
[2021-05-04 08:44:18,880] [1] [DEBUG] [Snowflake.Data.Client.SnowflakeDbConnection] Close Connection.
```
That is of course thrown by https://github.com/snowflakedb/snowflake-connector-net/blob/master/Snowflake.Data/Core/Authenticator/ExternalBrowserAuthenticator.cs#L171

--- a/Snowflake.Data/Snowflake.Data.csproj
+++ b/Snowflake.Data/Snowflake.Data.csproj
@@ -17,10 +17,6 @@
     <DebugType>Full</DebugType>
   </PropertyGroup>
   
-  <PropertyGroup Condition="$([System.Text.RegularExpressions.Regex]::IsMatch('$(TargetFramework)', '^net\d'))">
-	<DefineConstants>NETFRAMEWORK</DefineConstants>
-  </PropertyGroup>
-	
   <ItemGroup>
     <PackageReference Include="AWSSDK.S3" Version="3.7.0.4" />
     <PackageReference Include="Google.Cloud.Storage.V1" Version="2.5.0" />

